product/scanning.py

```diff
@@ -20,6 +21,11 @@
 from container.registry import retrieve_container_image
 from .model import ContainerImage, Component, UploadResult, UploadStatus
 
+class UploadMode(Enum):
+    UPDATE = 1
```

Comment:
I would prefer more expressive names (and potentially a _short_ (!) description of the semantics)

- UPDATE -> UPLOAD_IF_CHANGED (or UPLOAD_DIFF?)
- RESCAN -> "document: trigger rescan for all uploaded artifacts / images. (re-)upload absent artifacts"
- ALWAYS -> I'd prefer FORCE or FORCE_UPLOAD

Also, should this maybe be better named "ScanningMode", as the upload is just an implementation detail, whereas the actual semantics we are after is the scanning